application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
```

Comment:
I think your approach is certainly easier to read and understand. The test now seems non-racey as well.

The only real open question to me is whether we need the `atomicSwitch` for some reason. @gambol99, you created the code in the first place. Can you shed some light on whether @dmajere's implementation is equivalent to the previous code from your perspective?

Assuming it is, I have a few comments:

- You have to stop the Ticker in order to release the associated resources (see [docs](https://golang.org/pkg/time/#NewTicker)). So store the ticker in a variable, call `defer ticker.Stop()`, and read from the channel in the select statement.
- The documentation isn't clear on whether one needs to stop the timer as well. I'm thinking that you don't have to because the timer can clean up after itself once it has fired its single event. Not sure about this though.
- Assuming that we don't need to stop the timer: I think you can simplify a bit by dropping the `ticker` variable definition and replace `case <-ticker` with `case <-After(time.Millisecond * 500)`.